pallets/validators-manager/src/tests/tests.rs

```diff
@@ -588,7 +588,9 @@ mod add_validator {
                 let context = &AddValidatorContext::default();
 
                 set_session_keys(&context.collator);
-                <<ValidatorManager as Store>::ValidatorAccountIds>::append(&context.collator);
+                assert_ok!(<<ValidatorManager as Store>::ValidatorAccountIds>::try_append(
```

Comment:
Shouldn't we add a test in here for when we have the maximum number of collators and try to add beyond it?

Reply:
What is the behaviour the test is testing?
The BoundedVector implementation?

Reply:
No, actually not. We are doing functional testing, or in other words, behaviour-testing.

We are testing `add_validator`. If you see the sequence of contexts in the code, it would be something like this:

`add_validator fails_when validator_list_is_full`

So, you see, from the functionality's point of view, we don't care if we have bounded-vec or not. We now have a functional difference to what we had before, (a limit on addition) but we don't have that in the tests. In this case, the correct behaviour would be to receive a MaxValidators-type error.

It's a different function contract, so I think we should have a test to represent that.

Reply:
I see your point, although I think the test is of little value. We don't do such extensive testing on our endpoints.
The mock effort required trumps the value the test adds, as reaching this number of collator is academic.
The bound is not there to add such functionality to the system, but to align the implementation to the framework guidelines.
That being said I will add the test to accommodate the request as it probably won't be that complicated.
This won't be the case though for every bounded vector we convert.